javaagent-internal-logging-application/src/main/java/io/opentelemetry/javaagent/logging/application/ApplicationLoggerFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.logging.application;
+
+import io.opentelemetry.javaagent.bootstrap.ApplicationLoggerBridge;
+import io.opentelemetry.javaagent.bootstrap.InternalLogger;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+final class ApplicationLoggerFactory extends ApplicationLoggerBridge
+    implements InternalLogger.Factory {
+
+  private final AtomicBoolean installed = new AtomicBoolean();
+  private volatile InternalLogger.Factory actual = null;
+  private final ConcurrentMap<String, ApplicationLogger> inMemoryLoggers =
+      new ConcurrentHashMap<>();
+
+  private final InMemoryLogStore inMemoryLogStore;
+
+  ApplicationLoggerFactory(InMemoryLogStore inMemoryLogStore) {
+    this.inMemoryLogStore = inMemoryLogStore;
+  }
+
+  @Override
+  protected void install(InternalLogger.Factory applicationLoggerFactory) {
+    // just use the first bridge that gets discovered and ignore the rest
+    if (installed.compareAndSet(false, true)) {
```

Comment:
invert the`compareAndSet()` to return early (shrink the indent for the rest of the larger method).